crates/sui-config/src/genesis.rs

```diff
@@ -224,52 +205,191 @@ impl Builder {
+    // non-zero addresses, [`Transaction::input_objects_in_compiled_modules`] will consider
+    // them as dependencies even though they are not. Hence input_objects contain objects
+    // that don't exist on-chain because they are yet to be published.
+    #[cfg(debug_assertions)]
+    {
+        let to_be_published_addresses: HashSet<_> = modules
+            .iter()
+            .map(|module| *module.self_id().address())
+            .collect();
+        assert!(
+            // An object either exists on-chain, or is one of the packages to be published.
+            inputs
+                .iter()
+                .zip(input_objects.iter())
+                .all(|(kind, obj_opt)| obj_opt.is_some()
+                    || to_be_published_addresses.contains(&kind.object_id()))
+        );
+    }
+    let filtered = inputs
+        .into_iter()
+        .zip(input_objects.into_iter())
+        .filter_map(|(input, object_opt)| object_opt.map(|object| (input, object.to_owned())))
+        .collect::<Vec<_>>();
+
+    debug_assert!(ctx.digest() == TransactionDigest::genesis());
+    let mut temporary_store =
+        TemporaryStore::new(&store, InputObjects::new(filtered), ctx.digest());
+    let package_id = ObjectID::from(*modules[0].self_id().address());
+    let natives = native_functions.clone();
+    let mut gas_status = SuiGasStatus::new_unmetered();
+    let vm = adapter::verify_and_link(
+        &temporary_store,
+        &modules,
+        package_id,
+        natives,
+        &mut gas_status,
+    )?;
+    adapter::store_package_and_init_modules(
+        &mut temporary_store,
+        &vm,
+        modules,
+        ctx,
+        &mut gas_status,
+    )?;
+
+    let (_objects, _mutable_inputs, written, deleted, _events) = temporary_store.into_inner();
+
+    store.finish(written, deleted);
+
+    Ok(())
+}
+
+pub fn generate_genesis_system_object(
+    store: &mut InMemoryStorage,
+    move_vm: &MoveVM,
+    committee: &[ValidatorInfo],
+    genesis_ctx: &mut TxContext,
+) -> Result<()> {
+    let genesis_digest = genesis_ctx.digest();
+    let mut temporary_store =
+        TemporaryStore::new(&store, InputObjects::new(vec![]), genesis_digest);
+
+    let mut pubkeys = Vec::new();
+    let mut sui_addresses = Vec::new();
+    let mut network_addresses = Vec::new();
+    let mut names = Vec::new();
+    let mut stakes = Vec::new();
+
+    for (i, validator) in committee.iter().enumerate() {
+        pubkeys.push(validator.public_key());
+        sui_addresses.push(validator.sui_address());
+        network_addresses.push(validator.network_address());
+        // TODO: Allow config to specify human readable validator names.
+        names.push(format!("validator-{i}").into_bytes());
+        stakes.push(validator.stake());
+    }
+
+    adapter::execute(
+        move_vm,
+        &mut temporary_store,
+        ModuleId::new(SUI_FRAMEWORK_ADDRESS, ident_str!("genesis").to_owned()),
```

Comment:
probably fine, but would be nice to have these in a constant somewhere (we are very inconsistent about that for all the various module ID stuff and function name stuff, so not necessary for this PR)

Reply:
Yeah this was just a code move, I agree we could clean this up later